src/swell/tasks/run_jedi_variational_executable.py

```diff
@@ -162,8 +163,13 @@ def dict_to_ordereddict(d):
         # -----------------------
         if not generate_yaml_and_exit:
             self.logger.info('Running '+jedi_executable_path+' with '+str(np)+' processors.')
-            run_executable(self.logger, self.cycle_dir(), np, jedi_executable_path,
-                           jedi_config_file, output_log_file)
+            if perhost is None:
+               run_executable(self.logger, self.cycle_dir(), np, jedi_executable_path,
+                              jedi_config_file, output_log_file)
+            else:
+               perhost = eval(perhost)
+               run_executable(self.logger, self.cycle_dir(), np, jedi_executable_path,
+                              jedi_config_file, output_log_file, perhost)
```

Comment:
Your `run_executable` already handles the case of `perhost = None`, so you can simplify this:

```suggestion
               run_executable(self.logger, self.cycle_dir(), np, jedi_executable_path,
                              jedi_config_file, output_log_file, perhost)
```